server/src/controllers/auth.controller.js

```diff
@@ -1,106 +1,107 @@
-        phoneNumber:newUser.phoneNumber
-      });
-    }else{
-      res.status(400).json({error:'Invalid user data'})
+        _id: newUser._id,
+        email: newUser.email,
+        phoneNumber: newUser.phoneNumber
+      })
+    } else {
+      res.status(400).json({ error: 'Invalid user data' })
     }
-  }catch(error){
-    console.log('Error in register controller',error.message)
-    res.status(500).json({error:'Internal server error'})
+  } catch (error) {
+    console.log('Error in register controller', error.message)
+    res.status(500).json({ error: 'Internal server error' })
   }
 }
 
-export const login=async(req,res)=>{
-  try{
-    const {email,password}=req.body;
-    const user=await User.findOne({email});
-    const isPasswordCorrect=await bcrypt.compare(password, user.password || "");
-
-    if(!user || !isPasswordCorrect ){
-      return res.status(400).json({error:'Invalid email or password'})
+export const login = async (req, res) => {
+  try {
+    const { email, password } = req.body
+    const user = await User.findOne({ email })
+    const isPasswordCorrect = await bcrypt.compare(
+      password,
+      user.password || ''
+    )
+
+    if (!user || !isPasswordCorrect) {
+      return res.status(400).json({ error: 'Invalid email or password' })
     }
 
-    const token=generateTokenAndSetCookie(user._id,res)
+    const token = generateTokenAndSetCookie(user._id, res)
 
     res.status(200).json({
-      token:token,
-      _id:user._id,
-      email:user.email,
-      phoneNumber:user.phoneNumber
-    });
-
-  }catch(error){
-    console.log('Error in login controller',error.message)
-    res.status(500).json({error:'Internal server error'})
+      token,
+      _id: user._id,
+      email: user.email,
+      phoneNumber: user.phoneNumber
+    })
+  } catch (error) {
+    console.log('Error in login controller', error.message)
+    res.status(500).json({ error: 'Internal server error' })
   }
 }
 
-export const homePageLogin=async(req,res)=>{
-  try{
-    const {email}=req.body;
-    const user=await User.findOne({email});
+export const homePageLogin = async (req, res) => {
+  try {
+    const { email } = req.body
+    const user = await User.findOne({ email })
 
-    if(!user){
-      return res.status(400).json({error:'Invlaid email'})
+    if (!user) {
+      return res.status(400).json({ error: 'Invlaid email' })
     }
 
-    const token=generateTokenAndSetCookie(user._id,res)
+    const token = generateTokenAndSetCookie(user._id, res)
 
     res.status(200).json({
-      token:token,
-      _id:user._id,
-      email:user.email,
-      phoneNumber:user.phoneNumber
-    });
-
-  }catch(error){
-    console.log('Error in login controller',error.message)
-    res.status(500).json({error:'Internal server error'})
+      token,
+      _id: user._id,
+      email: user.email,
+      phoneNumber: user.phoneNumber
+    })
+  } catch (error) {
+    console.log('Error in login controller', error.message)
```

Comment:
Remove console log from backend and either just throw an exception or use some logger solution